src/hex.cpp

```diff
@@ -73,28 +73,28 @@ namespace libtorrent
 	extern char const hex_chars[];
 
 	char const hex_chars[] = "0123456789abcdef";
-
-	std::string to_hex(span<char const> s)
+	void to_hex(char const* in, size_t const len, char* out)
```

Comment:
if you intend to expose this function (i.e. give it external linkage / not put it in the anonymous namespace), I think you should also:

1. add a declaration of it to the header file
2. add a test case that exercises it in the test